src/LazyOperations/CartesianProduct.jl

```diff
@@ -295,6 +295,26 @@ function concretize(cp::CartesianProduct)
     return cartesian_product(concretize(cp.X), concretize(cp.Y))
 end
 
+function project(cp::CartesianProduct, block::AbstractVector{Int})
+    res = _project_cp_same_block(cp, block)
+    if res == nothing
+        res = invoke(project, Tuple{LazySet, typeof(block)}, cp, block)
```

Comment:
use an auxilary function?  `invoke`  makes it less clear the function that is actually used (it may have an impact in performance too?)